src/components/common/PeopleAndRoles/OrgPerson.vue

```diff
@@ -371,82 +414,171 @@ export default class OrgPerson extends Mixins(CommonMixin) {
+  /**
+   * The delivery address schema for the current org-person.
+   * See also PeopleAndRoles.vue::haveRequiredAddresses.
+   */
+  get deliveryAddressSchema (): AddressSchemaIF {
+    // proprietor/partner delivery address must be in BC, Canada
+    if (this.isProprietor || this.isPartner) return InBcCanadaAddressSchema
 
-    if ((this.isDirector || this.isProprietor || this.isPartner) && !this.inheritMailingAddress) {
-      isFormValid = (isFormValid && this.deliveryAddressValid)
-    }
+    // directors delivery address can be anywhere in the world
+    if (this.isDirector) return DefaultAddressSchema
 
-    if (this.isChangeFiling && (this.isProprietor || this.isPartner)) {
-      if (this.hasOrgPersonNameChanged(this.orgPerson)) {
-        isFormValid = (isFormValid && this.orgPerson.confirmNameChange)
-      }
-    }
+    // other persons delivery address can be anywhere in the world
+    if (this.isPerson) return DefaultAddressSchema
 
-    return isFormValid
+    // other orgs delivery address must be in BC, Canada
+    return InBcCanadaAddressSchema
   }
 
-  /** True if current data object is a person. */
-  get isPerson (): boolean {
-    return (this.orgPerson?.officer.partyType === PartyTypes.PERSON)
-  }
+  /** Whether to disable the "same as" checkbox (to force entry of delivery address). */
+  get disableSameDeliveryAddress (): boolean {
+    // proprietor/partner delivery address must be in BC, Canada
+    if (this.isProprietor || this.isPartner) {
+      return (
+        this.inProgressMailingAddress.addressRegion !== REGION_BC ||
+        this.inProgressMailingAddress.addressCountry !== COUNTRY_CA
+      )
+    }
 
-  /** True if current data object is an organization (corporation/firm). */
-  get isOrg (): boolean {
-    return (this.orgPerson?.officer.partyType === PartyTypes.ORGANIZATION)
+    // directors delivery address can be anywhere in the world
+    if (this.isDirector) return false
+
+    // other persons delivery address can be anywhere in the world
+    if (this.isPerson) return false
+
+    // other orgs delivery address must be in BC, Canada
+    return (
+      this.inProgressMailingAddress.addressRegion !== REGION_BC ||
+      this.inProgressMailingAddress.addressCountry !== COUNTRY_CA
+    )
   }
 
-  get orgTypesLabel (): string {
-    return this.getResource.changeData.orgPersonInfo?.orgTypesLabel
+  /** Whether the Remove button should be rendered. */
+  get showRemoveBtn (): boolean {
+    switch (true) {
+      case (this.isAlterationFiling): {
+        // alterations don't use this component
+        return false
+      }
+      case (this.isChangeRegFiling): {
+        // can only remove partner in a change of registration filing
+        return this.isPartner
+      }
+      case (this.isConversionFiling): {
+        return true
+      }
+      case (this.isCorrectionFiling): {
+        return true
+      }
+    }
+    return false // should never happen
   }
```

Comment:
This getter is a good example of "centralized rules" for all filing types. I have added a few of these in this component to greatly simplify the logic around when things are allowed, or what schema to use, etc.